src/server/tiered_storage.h

```diff
@@ -78,19 +78,34 @@ class TieredStorage {
   // Run offloading loop until i/o device is loaded or all entries were traversed
   void RunOffloading(DbIndex dbid);
 
+  size_t ReclaimMemory(size_t goal);
+
+  size_t CoolMemoryUsage() const {
+    return cool_memory_used_;
+  }
+
  private:
   // Returns if a value should be stashed
   bool ShouldStash(const PrimeValue& pv) const;
 
+  // Moves pv contents to the cool storage and updates pv to point to it.
+  void CoolDown(DbIndex db_ind, std::string_view str, const tiering::DiskSegment& segment,
+                PrimeValue* pv);
+
   // Returns the primary value, and deletes the cool item as well as its offloaded storage.
   PrimeValue Warmup(DbIndex dbid, PrimeValue::CoolItem item);
 
+  PrimeValue DeleteCool(detail::TieredColdRecord* record);
+  detail::TieredColdRecord* PopCool();
+
   PrimeTable::Cursor offloading_cursor_{};  // where RunOffloading left off
 
   std::unique_ptr<ShardOpManager> op_manager_;
   std::unique_ptr<tiering::SmallBins> bins_;
-  CoolQueue cool_queue_;
+  typedef ::boost::intrusive::list<detail::TieredColdRecord> CoolQueue;
```

Comment:
Sometimes when you push lots of code you get stuck in c++03 mode 😆

Reply:
`typedef` is the cool kids keyword, don't you know?